libvuln/http/updatediff.go

```diff
@@ -0,0 +1,95 @@
+package http
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"path"
+
+	"github.com/google/uuid"
+
+	"github.com/quay/claircore/libvuln"
+)
+
+func fail(w http.ResponseWriter, err error) {
+	w.WriteHeader(http.StatusInternalServerError)
+	// Nothing to be done for a write error.
+	if err := json.NewEncoder(w).Encode(struct {
+		Error error `json:"error"`
+	}{
+		Error: err,
+	}); err != nil {
+		panic(err)
+	}
+}
+
+// UpdateDiffs assumes it's anchored to a path segment ending in "/".
+func UpdateDiffs(l *libvuln.Libvuln) http.Handler {
```

Comment:
Wouldn't it make sense to split this out?

```
const (
  updatesPath = "api/v1/updates"
  updatesDiff = "api/v1/updates/diff"
)

func UpdatesHandler() http.Handlerfunc {
  do the only thing you do when you go to the root of updates
}
func UpdateDiffsHandler() http.HandlerFunc {
  return func(w http.ResponseWriter, r *http.Request) {
     switch r.Method {
     case http.MethodGet:
       do your get stuff
     case http.MethodPut:
       do your put stuff
     case http.Delete:
       do your delete stuff
```

Now as *soon* as I look at the handler, I know what code is running for what http method.